### PR TITLE
Apply setting values immediately

### PR DESCRIPTION
**Pull request by eMazeika**
https://virtocommerce.atlassian.net/browse/VP-2350

### CODE REVIEW

**Comment by lnetrebskii** [comment]
Review task https://virtocommerce.atlassian.net/browse/VP-2374 has been created

File: src/VirtoCommerce.CatalogPersonalizationModule.Data/Common/ModuleConfigurator.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                var cronExpression = _settingsManager.GetValue(Settings.General.CronExpression.Name, "0/15 * * * *");
+                RecurringJob.AddOrUpdate<TaggedItemOutlinesSynchronizationJob>(TaggedItemOutlinesSynchronizationJob.JobId, x => x.Run(), cronExpression);
+            }
+            else
+            {
+                RecurringJob.RemoveIfExists(TaggedItemOutlinesSynchronizationJob.JobId);
+            }
+        }
+
+        public void ConfigureSearch()
```
[the remark this answers is not in the document]
Comment (eMazeika) [reply src/VirtoCommerce.CatalogPersonalizationModule.Data/Common/ModuleConfigurator.cs:45]: @mvktsk , can you suggest a way to "reduce Cognitive Complexity"?

File: src/VirtoCommerce.CatalogPersonalizationModule.Data/Common/ModuleConfigurator.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+using System.Threading.Tasks;
+using Hangfire;
+using VirtoCommerce.CatalogPersonalizationModule.Data.BackgroundJobs;
+using VirtoCommerce.CatalogPersonalizationModule.Data.Search.Indexing;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Core.Settings;
+using VirtoCommerce.SearchModule.Core.Model;
+using static VirtoCommerce.CatalogPersonalizationModule.Core.ModuleConstants;
+
+namespace VirtoCommerce.CatalogPersonalizationModule.Data.Common
```
Comment (tatarincev) [inline src/VirtoCommerce.CatalogPersonalizationModule.Data/Common/ModuleConfigurator.cs:12]: Regarding these changes, we have some vision of how to index configuration should look like, we need to discuss before you can start the concrete implementation of this improvement.

Comment (eMazeika) [reply src/VirtoCommerce.CatalogPersonalizationModule.Data/Common/ModuleConfigurator.cs:12]: Separate task created https://virtocommerce.atlassian.net/browse/VP-2386

**Comment by eMazeika** [comment]
> See my comments [VirtoCommerce/vc-module-subscription#33](https://github.com/VirtoCommerce/vc-module-subscription/pull/33)

Separate task created https://virtocommerce.atlassian.net/browse/VP-2385

**Comment by eMazeika** [comment]
@yecli , any comments?

**Comment by yecli** [comment]
Probably it was closed automatically when I accidentally removed the origin/dev-3.0.0(!) by pressing Del branch from VS.
Need to add protection.
Returned branch and reopened PR.

File: src/VirtoCommerce.CatalogPersonalizationModule.Data/Common/ModuleConfigurator.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        public ModuleConfigurator(ISettingsManager settingsManager, IEnumerable<IndexDocumentConfiguration> documentIndexingConfigurations, TaggedItemIndexChangesProvider changesProvider, CategoryTaggedItemDocumentBuilder categoryDocumentBuilder, ProductTaggedItemDocumentBuilder productDocumentBuilder)
+        {
+            _settingsManager = settingsManager;
+            _documentIndexingConfigurations = documentIndexingConfigurations;
+            _changesProvider = changesProvider;
+            _categoryDocumentBuilder = categoryDocumentBuilder;
+            _productDocumentBuilder = productDocumentBuilder;
+        }
+
+        public async Task ConfigureOutlinesSynchronizationJob()
```
Comment (akak1977) [inline src/VirtoCommerce.CatalogPersonalizationModule.Data/Common/ModuleConfigurator.cs:31]: It seems there is the similar code in changes for each module.
I think we should write it once in platform.

**Pull request closed without merging**